src/domain/usecases/SaveDataSetUseCase.ts

```diff
@@ -0,0 +1,39 @@
+import _ from "$/domain/entities/generic/Collection";
+import { DataSet } from "$/domain/entities/DataSet";
+import { Future, FutureData } from "$/domain/entities/generic/Future";
+import { DataSetRepository } from "$/domain/repositories/DataSetRepository";
+import { Maybe } from "$/utils/ts-utils";
+import { getErrors } from "$/domain/entities/generic/Error";
+
+export class SaveDataSetUseCase {
+    constructor(private dataSetRepository: DataSetRepository) {}
+
+    execute(dataSet: DataSet): FutureData<void> {
+        return this.getDataSetById(dataSet.id).flatMap(existingDataSet => {
+            const dataSetToSave = DataSet.create({
+                ...(existingDataSet || {}),
+                ...dataSet,
+                shortName: this.truncateValue(dataSet.name),
+            });
+
+            const result = dataSetToSave.validateSetup();
+            if (result.isError()) {
+                const errors = getErrors(result.value.error);
+                return Future.error(new Error(errors.join("\n")));
+            }
+
+            return this.dataSetRepository.save([dataSetToSave]);
```

Comment:
Not sure about this. For me, `this.dataSetRepository.save(dataSet)` should take care of get the existing + merge + save. Unless the domain needs to interact somehow with the existing dataSet, but I think the validation should be done over dataSet.